pkg/virt-operator/resource/generate/components/secrets.go

```diff
@@ -464,49 +465,60 @@ func LoadCertificates(secret *k8sv1.Secret) (serverCrt *tls.Certificate, err err
 	return &crt, nil
 }
 
+func filterValidCertificates(certs []*x509.Certificate, now time.Time) []*x509.Certificate {
```

Comment:
Because `sortCertificatesByAge` also has side effects on the passed in `certs` slice, would it potentially make sense to delete expired certs in this helper instead of creating a new slice?